### PR TITLE
feat/MS-5805_continue promise execution if contact or service cannot …

### PR DESCRIPTION
**Pull request by SkiiNox**
…be decrypted

### CODE REVIEW

File: icc-x-api/icc-contact-x-api.ts [hunk 1]
```diff
@@ -565,7 +565,10 @@ export class IccContactXApi extends iccContactApi {
                                   console.log("Cannot decrypt service", svc.id)
                                   resolve(null)
                                 }
-                              )
+                              ).catch(e => {
+                                console.log("Cannot decrypt service", svc.id);
+                                return null;
```
Comment (fabien-zimmer) [inline icc-x-api/icc-contact-x-api.ts:570]: Should be resolve(null); instead of return null;

File: icc-x-api/icc-contact-x-api.ts [hunk 2]
```diff
@@ -592,8 +595,14 @@ export class IccContactXApi extends iccContactApi {
                                     console.log("Cannot decrypt service", svc.id)
                                     resolve(null)
                                   }
-                                )
+                                ).catch(e => {
+                                  console.log("Cannot decrypt service", svc.id);
+                                  return null;
```
Comment (fabien-zimmer) [inline icc-x-api/icc-contact-x-api.ts:600]: Should be resolve(null); instead of return null;

File: icc-x-api/icc-contact-x-api.ts [hunk 3]
```diff
@@ -638,10 +647,16 @@ export class IccContactXApi extends iccContactApi {
                                 console.log("Cannot decrypt contact", ctc.id)
                                 resolve(ctc)
                               }
-                            )
+                            ).catch(e => {
+                              console.log("Cannot decrypt contact", ctc.id);
+                              return ctc;
```
Comment (fabien-zimmer) [inline icc-x-api/icc-contact-x-api.ts:652]: Should be resolve(ctc); instead of return ctc;

File: icc-x-api/icc-contact-x-api.ts (hunk 1 above)
Comment (jeanbaptistedeborman) [inline icc-x-api/icc-contact-x-api.ts]: - `return null;` probably unnecessary as the function will return falsy `undefined `anymay.
- avoid console.log. Prefer `console.warn()` or LoggerService.

**Pull request merged**